analyzers/src/SonarAnalyzer.Common/Rules/ToStringShouldNotReturnNullBase.cs

```diff
@@ -63,5 +65,6 @@ protected void ToStringReturnsNull(SyntaxNodeAnalysisContext context, SyntaxNode
         node.Ancestors()
             .TakeWhile(x => !IsLocalOrLambda(x))
             .Any(x => Language.Syntax.IsKind(x, MethodKind)
-                && nameof(ToString).Equals(Language.Syntax.NodeIdentifier(x)?.ValueText, Language.NameComparison));
+                && nameof(ToString).Equals(Language.Syntax.NodeIdentifier(x)?.ValueText, Language.NameComparison)
+                && !IsStatic(x));
```

Comment:
As the case makes sense for both C# and VB.NET. I suggest to add a helper method also for VB.NET to check if a sub or function is static/shared. And after add a method to Language.Syntax (IsStatic) that will call the checks for C# and VB.NET and will return true if a method or sub or function is static/shared. This way you should be able to do the call from here and you would not need the abstract IsStatic method.